scrapy/exporters.py

```diff
@@ -175,17 +175,22 @@ class CsvItemExporter(BaseItemExporter):
 
     def __init__(self, file, include_headers_line=True, join_multivalued=',', **kwargs):
         self._configure(kwargs, dont_fail=True)
-        if not self.encoding:
-            self.encoding = 'utf-8'
-        self.include_headers_line = include_headers_line
+        self.encoding = self.encoding or 'utf-8'
         self.stream = io.TextIOWrapper(
             file,
             line_buffering=False,
             write_through=True,
             encoding=self.encoding
         ) if six.PY3 else file
+
+        # don't write a header to a file with contents
+        self.stream.seek(0, 2)
+        if self.stream.tell() > 0:
```

Comment:
Previously exporters didn't require .seek or .tell methods to be implemented. I'm not sure it works e.g. with StdoutFeedStorage.

I think this is a great feature, but autodetection can be tricky. 
What do you think about adding an option to disable CSV headers instead? It won't be as nice and easy to use, but user will have a full control on it.